core/src/main/scala/org/apache/spark/sql/SnappyBaseParser.scala

```diff
@@ -116,7 +116,7 @@ abstract class SnappyBaseParser(session: SnappySession) extends Parser {
   protected def start: Rule1[LogicalPlan]
 
   protected final def identifier: Rule1[String] = rule {
-    atomic(capture(CharPredicate.Alpha ~ Consts.identifier.*)) ~
+    atomic(capture((CharPredicate("_") | CharPredicate.Alpha) ~ Consts.identifier.*)) ~
```

Comment:
For a single char just '_' without CharPredicate is more efficient. However, for best performance it is better to declare a new CharPredicate in SnappyParserConsts that adds '_' to Alpha like Consts.identifier: CharPredicate.Alpha ++ CharPredicate('_')